src/constructs/ec2/security-groups.ts

```diff
@@ -1,42 +1,60 @@
 import type { CfnSecurityGroup, IPeer, SecurityGroupProps } from "@aws-cdk/aws-ec2";
 import { Peer, Port, SecurityGroup } from "@aws-cdk/aws-ec2";
-import { transformToCidrIngress } from "../../utils/security-groups";
 import type { GuStack } from "../core";
 
-export interface CidrIngress {
+/**
+ * A way to describe an ingress or egress rule for a security group.
+ *
+ * See [[transformToSecurityGroupAccessRule]] for a handy helper function.
+ */
+export interface SecurityGroupAccessRule {
+  /**
+   * The CIDR address for this rule.
+   * Use `Peer.anyIpv4()` for global access.
+   */
   range: IPeer;
-  description: string;
-}
 
-export interface CidrEgress {
-  range: IPeer;
-  port: Port;
-  description?: string;
+  /**
+   * The port to open in a security group.
+   * The default protocol is TCP.
+   * Use `Port.udp(port)` for the UDP protocol.
+   */
+  port: number | Port;
+
+  /**
+   * A short explanation for this rule.
+   */
+  description: string;
 }
 
 export interface GuSecurityGroupProps extends SecurityGroupProps {
   overrideId?: boolean;
-  ingresses?: CidrIngress[];
-  egresses?: CidrEgress[];
+  ingresses?: SecurityGroupAccessRule[];
+  egresses?: SecurityGroupAccessRule[];
 }
 
 export class GuSecurityGroup extends SecurityGroup {
-  static defaultIngressPort = Port.tcp(443);
-
   constructor(scope: GuStack, id: string, props: GuSecurityGroupProps) {
     super(scope, id, props);
 
     if (props.overrideId) {
       (this.node.defaultChild as CfnSecurityGroup).overrideLogicalId(id);
     }
 
-    props.ingresses?.forEach(({ range, description }) =>
-      this.addIngressRule(range, GuSecurityGroup.defaultIngressPort, description)
```

Comment:
The ingress rule port was always set to port 443, which is a bug, I think?!

Reply:
As we are already providing the HTTPS default via `GuPublicInternetAccessSecurityGroup`, I think allowing more flexibility at the `GuSecurityGroup` level is fine.

If we do want to be stricter, we could follow the example you've set for port 22 and explicitly throw if users attempt to allow incoming traffic on port 80?

Reply:
There might be legit use cases to allow incoming on port 80 - if your server is running on EC2 on 80, you'd want the load balancer to connect, but nothing else. Not sure if can encode that cleanly yet? Maybe it can be a warning that's printed out?

Reply:
Good point, let's leave it for now?